app/components/goodcity/designate-form.js

```diff
@@ -52,5 +66,33 @@ export default Ember.Component.extend({
 
       return this.get("packageHasOtherDesignations");
     }
-  )
+  ),
+
+  actions: {
+    displayItemAvailabilityOverlay() {
+      this.set("showItemAvailability", true);
+    },
+
+    closeItemAvailabilityOverlay() {
+      this.set("showItemAvailability", false);
+    },
+
+    gainItem() {
+      this.set("showItemAvailability", false);
+      this.set("readyToDesignate", false);
+      this.router.transitionTo("items.detail.publishing", this.get("pkg.id"));
+    },
+
+    modifyDesignation() {
+      this.set("showItemAvailability", false);
+      this.set("readyToDesignate", false);
+      this.router.transitionTo("items.detail.publishing", this.get("pkg.id"));
+    },
+
+    modifyBoxPalletAllocation() {
+      this.set("showItemAvailability", false);
+      this.set("readyToDesignate", false);
+      this.router.transitionTo("items.detail.location", this.get("pkg.id"));
+    }
```

Comment:
can it be done something like this? since only the routing changes and rest all are same

```suggestion
    action(name){
        this.set("showItemAvailability", false);
        this.set("readyToDesignate", false);
        switch(name){
           case 'gain':
              this.router.transitionTo("items.detail.publishing", this.get("pkg.id"));
              break;
           case 'modifyDesignation':
              this.router.transitionTo("items.detail.publishing", this.get("pkg.id"));
              break;
           case 'modifyBoxPalletAllocation':
             this.router.transitionTo("items.detail.location", this.get("pkg.id"));
             break;
        }
    
   
```